rbtree.go

```diff
@@ -133,7 +133,7 @@ func (t *Rbtree) rightRotate(x *Node) {
 	x.Parent = y
 }
 
-func (t *Rbtree) insert(z *Node) {
+func (t *Rbtree) insert(z *Node) Item {
```

Comment:
I see your point and I agree with you.  Everything in there is just dealing with Nodes so we should keep it consistent.  I will update it tonight sometime.